### ISSUE TITLE
Focus is lost on shipping selection after toggling between available and disabled.

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#48104]
Currently, we only disable radio inputs when making network requests, this causes focus to be lost when making a network request, for example selecting a shipping rate. Focus should be retained on that element.

It appears that there is no definitive decision around this, check this Gutenberg issue about that https://github.com/WordPress/gutenberg/issues/56547

Given that, I think we should retain focus on our disabled radio inputs, unless we know for sure other cases in which a radio input will stay visible but disabled.

### Steps to reproduces

1. Have at least 2 rates.
2. On Checkout, click the first shipping element.
3. Using keyboard, select the second option.
4. Wait until network request is done.
5. Click arrow up, notice that the page scrolls up instead of selecting the top element.

### Expexted behavior

Focus should be retained and narrowing up should select the previous element.

**Comment by mikejolley** [comment]
Update as I've been digging into this:

1. Its not so simple to just disable `disabled` here because we make a request each time an option is toggled. This causes multiple requests and the UI to update chaotically.
2. We could look into request cancelling (see refs below) but so far I've found it can still act strange if a request succeeds just as you toggle. Do we need more local state?
3. Alternatively we could look into disabling navigation while it loads, more debouncing, or restoring focus after load. These may lead to a slow experience however when navigating through several items in the list. 

Refs:
- https://github.com/woocommerce/woocommerce-blocks/compare/trunk...try/cancel-request-middleware#diff-0e298e04c720d10b99ec570baf747e632a4423f62c9e867d3bf81d9241886a82
- https://github.com/WordPress/gutenberg/blob/bd6e764e1ba988d35ab17b5cfcb678715051fdd0/packages/api-fetch/README.md?plain=1#L71
